java/core/src/test/java/sleeper/core/schema/SchemaTest.java

```diff
@@ -57,48 +61,132 @@ public void equalsShouldReturnCorrectResults() {
+                new Field("column1", new IntType()),
+                new Field("column2", new StringType()));
         Field refuseField = new Field("column3", new ListType(new IntType()));
+        Schema.Builder builder = Schema.builder().rowKeyFields(rowKeyFields).sortKeyFields(refuseField);
+        Schema schema = new Schema();
+        schema.setRowKeyFields(rowKeyFields);
 
         // When / Then
-        assertThatThrownBy(() -> schema1.setSortKeyFields(refuseField))
+        assertThatThrownBy(() -> schema.setSortKeyFields(refuseField))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContainingAll("Sort key", "type");
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContainingAll("Sort key", "type");
+    }
+
+    @Test
+    public void refuseNoKeyFields() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields()
+                .sortKeyFields(new Field("column1", new IntType()))
+                .valueFields(new Field("column2", new StringType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
                 .isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    public void refuseNullKeyFields() {
+        // Given
+        Schema.Builder builder = Schema.builder()
+                .rowKeyFields((List<Field>) null)
+                .sortKeyFields(new Field("column1", new IntType()))
+                .valueFields(new Field("column2", new StringType()));
+
+        // When / Then
+        assertThatThrownBy(builder::build)
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void setNoSortKeysByDefault() {
+        // Given
+        Field rowKeyField = new Field("column1", new IntType());
+        Field valueField = new Field("column2", new StringType());
+
+        // When / Then
+        assertThat(Schema.builder()
+                .rowKeyFields(rowKeyField).valueFields(valueField)
+                .build())
+                .isEqualTo(Schema.builder()
+                        .rowKeyFields(rowKeyField).valueFields(valueField)
+                        .sortKeyFields(Collections.emptyList())
+                        .build());
+    }
+
+    @Test
+    public void setNoValueFieldsByDefault() {
+        // Given
+        Field rowKeyField = new Field("column1", new IntType());
+        Field sortKeyField = new Field("column2", new StringType());
+
+        // When / Then
+        assertThat(Schema.builder()
+                .rowKeyFields(rowKeyField).sortKeyFields(sortKeyField)
+                .build())
+                .isEqualTo(Schema.builder()
+                        .rowKeyFields(rowKeyField).sortKeyFields(sortKeyField)
+                        .valueFields(Collections.emptyList())
+                        .build());
+    }
```

Comment:
Can we add a test setNoSortAndValueFieldsByDefault ? (It would be a bit odd, but it is allowed.)

Reply:
Done